dbms/src/Debug/AstToExecutor.cpp

```diff
@@ -0,0 +1,1516 @@
+        case tipb::ScalarFuncSig::CastIntAsReal:
+        case tipb::ScalarFuncSig::CastRealAsReal:
+        {
+            ci.tp = TiDB::TypeDouble;
+            break;
+        }
+        case tipb::ScalarFuncSig::RoundInt:
+        case tipb::ScalarFuncSig::RoundWithFracInt:
+        {
+            ci.tp = TiDB::TypeLongLong;
+            if (it_sig->first.find("uint") != std::string::npos)
+                ci.flag = TiDB::ColumnFlagUnsigned;
+            break;
+        }
+        case tipb::ScalarFuncSig::RoundDec:
+        case tipb::ScalarFuncSig::RoundWithFracDec:
+        {
+            ci.tp = TiDB::TypeNewDecimal;
+            break;
+        }
+        case tipb::ScalarFuncSig::RoundReal:
+        case tipb::ScalarFuncSig::RoundWithFracReal:
+        {
+            ci.tp = TiDB::TypeDouble;
+            break;
+        }
+        default:
+            ci.tp = TiDB::TypeLongLong;
+            ci.flag = TiDB::ColumnFlagUnsigned;
+            break;
+        }
+        for (const auto & child_ast : func->arguments->children)
+        {
+            compileExpr(input, child_ast);
+        }
+    }
+    else if (ASTLiteral * lit = typeid_cast<ASTLiteral *>(ast.get()))
+    {
+        switch (lit->value.getType())
+        {
+        case Field::Types::Which::Null:
+            ci.tp = TiDB::TypeNull;
+            // Null literal expr doesn't need value.
+            break;
+        case Field::Types::Which::UInt64:
+            ci.tp = TiDB::TypeLongLong;
+            ci.flag = TiDB::ColumnFlagUnsigned;
+            break;
+        case Field::Types::Which::Int64:
+            ci.tp = TiDB::TypeLongLong;
+            break;
+        case Field::Types::Which::Float64:
+            ci.tp = TiDB::TypeDouble;
+            break;
+        case Field::Types::Which::Decimal32:
+        case Field::Types::Which::Decimal64:
+        case Field::Types::Which::Decimal128:
+        case Field::Types::Which::Decimal256:
+            ci.tp = TiDB::TypeNewDecimal;
+            break;
+        case Field::Types::Which::String:
+            ci.tp = TiDB::TypeString;
+            break;
+        default:
+            throw Exception(String("Unsupported literal type: ") + lit->value.getTypeName(), ErrorCodes::LOGICAL_ERROR);
+        }
+    }
+    else
+    {
+        /// not supported unless this is a literal
+        throw Exception("Unsupported expression " + ast->getColumnName(), ErrorCodes::LOGICAL_ERROR);
+    }
+    return ci;
+}
+
+void compileFilter(const DAGSchema & input, ASTPtr ast, std::vector<ASTPtr> & conditions)
+{
+    if (auto * func = typeid_cast<ASTFunction *>(ast.get()))
+    {
+        if (func->name == "and")
+        {
+            for (auto & child : func->arguments->children)
+            {
+                compileFilter(input, child, conditions);
+            }
+            return;
+        }
+    }
+    conditions.push_back(ast);
+    compileExpr(input, ast);
+}
+} // namespace
+
+String LOCAL_HOST = "127.0.0.1:3930";
```

Comment:
move to anonymous namespace?

Reply:
done. I have moved it into namespace db::debug.